From oneukum@suse.de Tue Jun  5 01:51:01 2007
From: Oliver Neukum <oneukum@suse.de>
To: gregkh@suse.de, linux-usb-devel@lists.sourceforge.net,
	Andrew Morton <akpm@linux-foundation.org>,
	Sumant Patro <sumant.patro@lsi.com>,
	Seokmann Ju <seokmann.ju@lsil.com>
Subject: USB: io_ti: sleep with spinlock held detected by automatic tool
Date: Tue, 5 Jun 2007 10:50:48 +0200
Content-Disposition: inline
Message-Id: <200706051050.49439.oneukum@suse.de>

this fixes the sleep found with the automatic tool.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/io_ti.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/io_ti.c
+++ b/drivers/usb/serial/io_ti.c
@@ -2364,7 +2364,7 @@ static int restart_read(struct edgeport_
 		urb->complete = edge_bulk_in_callback;
 		urb->context = edge_port;
 		urb->dev = edge_port->port->serial->dev;
-		status = usb_submit_urb(urb, GFP_KERNEL);
+		status = usb_submit_urb(urb, GFP_ATOMIC);
 	}
 	edge_port->ep_read_urb_state = EDGE_READ_URB_RUNNING;
 	edge_port->shadow_mcr |= MCR_RTS;
